src/EnergyPlus/OutputProcessor.cc

```diff
@@ -6468,26 +6466,37 @@ void UpdateMeterReporting(EnergyPlusData &state)
     }
 
     // Helper lambda to locate a meter index from its name. Returns a negative value if not found
-    auto findMeterIndexFromMeterName = [&state](std::string const &name) -> int {
-        auto &op(state.dataOutputProcessor);
+    auto setupMeterFromMeterName =
+        [&state](std::string &name, std::string const &freqString, bool MeterFileOnlyIndicator, bool CumulativeIndicator) -> bool {
+        bool result = false;
 
-        // Return a value <= 0 if not found
-        int meterIndex = -99;
+        auto varnameLen = index(name, '[');
+        if (varnameLen != std::string::npos) {
+            name.erase(varnameLen);
+        }
+
+        auto &op(state.dataOutputProcessor);
 
         std::string::size_type wildCardPosition = index(name, '*');
 
         if (wildCardPosition == std::string::npos) {
-            meterIndex = UtilityRoutines::FindItem(name, op->EnergyMeters);
+            int meterIndex = UtilityRoutines::FindItem(name, op->EnergyMeters);
+            if (meterIndex > 0) {
+                ReportingFrequency ReportFreq = determineFrequency(state, freqString);
+                SetInitialMeterReportingAndOutputNames(state, meterIndex, MeterFileOnlyIndicator, ReportFreq, CumulativeIndicator);
+                result = true;
+            }
```

Comment:
Whereas before you would just return the found index, you now call `SetInitialMeterReportingAndOutputNames`.  I don't know the full detail of that function right now, but I'm assuming previously the calling code would need to call it after using the lambda to look up the index.  Seems fine so far.